readthedocs/builds/version_slug.py

```diff
@@ -30,7 +30,7 @@
 #   +? -- allow multiple of those, but be not greedy about the matching
 #   (?: ... ) -- wrap everything so that the pattern cannot escape when used in
 #                regexes.
-VERSION_SLUG_REGEX = '(?:[a-z0-9][-._a-z0-9]+?)'
+VERSION_SLUG_REGEX = '(?:[a-z0-9][-._a-z0-9]*?)'
```

Comment:
I believe the final `?` is not required here. `*` should catch none/any number of instances of allowed characters.